driven/storage/adapter.go

```diff
@@ -847,6 +847,38 @@ func (sa Adapter) DeleteMessageWithContext(ctx context.Context, orgID string, ap
 	return nil
 }
 
+// UpdateUnreadMessage updates a unread message in the recipients to read
+func (sa Adapter) UpdateUnreadMessage(message *model.Message, userID *string) (*model.Message, error) {
+	if message != nil {
+		updateRecipients := []model.Recipient{}
+		for _, recipient := range message.Recipients {
+			if *userID == *recipient.UserID {
+				if recipient.Read == false {
```

Comment:
This iteration is not needed at all. This oparation can be achieved with a single  mongo operation. Example is already given and it partially is used within the code below.